controllers/front/OrderDetailController.php

```diff
@@ -90,7 +90,8 @@ public function postProcess()
 
                     $cm->id_customer_thread = $ct->id;
                     $cm->message = $msgText;
-                    $cm->ip_address = (int)ip2long($_SERVER['REMOTE_ADDR']);
+                    $client_ip_address = Tools::getRemoteAddr();
+                    $cm->ip_address = (int)ip2long($client_ip_address);
```

Comment:
Okay for using Tools::getRemoteAddr(), but why using ip2long? I'm pretty sure we never do the opposite operation in the whole project.